sdk/communication/azure-communication-mediacomposition/src/main/java/com/azure/communication/mediacomposition/MediaCompositionClient.java

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.communication.mediacomposition;
+
+import com.azure.communication.mediacomposition.implementation.MediaCompositionsImpl;
+import com.azure.communication.mediacomposition.implementation.models.CompositionStreamState;
+import com.azure.communication.mediacomposition.implementation.models.MediaCompositionBody;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.http.rest.Response;
+import com.azure.core.util.Context;
+
+import java.util.Objects;
+
+/**
+ * Synchronous client for Communication service media composition operations
+ */
+@ServiceClient(builder = MediaCompositionClientBuilder.class, isAsync = false)
+public final class MediaCompositionClient {
+
+    private final MediaCompositionsImpl svcImpl;
+
+    MediaCompositionClient(MediaCompositionsImpl svcImpl) {
+        this.svcImpl = svcImpl;
+    }
+
+    /**
+     * Get MediaComposition by Id
+     * @param mediaCompositionId The mediaCompositionId parameter.
+     * @return {@link MediaCompositionBody}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public MediaCompositionBody getMediaComposition(String mediaCompositionId) {
+        Objects.requireNonNull(mediaCompositionId, "'mediaCompositionId' cannot be null.");
+        return svcImpl.get(mediaCompositionId);
+    }
+
+    /**
+     * Get MediaComposition by Id
+     * @param mediaCompositionId The mediaCompositionId parameter.
+     * @param context The context to associate with this operation.
+     * @return {@link Response} of {@link MediaCompositionBody}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<MediaCompositionBody> getMediaCompositionWithResponse(String mediaCompositionId, Context context) {
+        Objects.requireNonNull(mediaCompositionId, "'mediaCompositionId' cannot be null.");
+        context = context == null ? Context.NONE : context;
+        return svcImpl.getWithResponseAsync(mediaCompositionId, context).block();
+    }
+
+    /**
+     * Create a MediaComposition with data
+     * @param mediaCompositionBody The MediaCompositionBody parameter.
+     * @return {@link MediaCompositionBody}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public MediaCompositionBody createMediaComposition(MediaCompositionBody mediaCompositionBody) {
+        Objects.requireNonNull(mediaCompositionBody, "'mediaCompositionBody' cannot be null");
+        Objects.requireNonNull(mediaCompositionBody.getId(), "'mediaCompositionBody' must contain a valid Id");
```

Comment:
We should not be validating server-side properties in the client. The general guideline is that we should only validate client side properties. So for example, mediaCompositionBody should be validated in the client side because we use `mediaCompositionBody.getId()` but the id itself should be validated by the service

Reply:
Same with the other methods in the client

Reply:
Id is a grey/boundary case. On client side, we use Id to create the URL before it hits the service. If the Id is null, the url will be invalid. IMHO, we'd better fail early before a 400 is returned (without even hitting service at all which can be really confusing).

Reply:
I see, that makes sense to me :)